app/jobs/assignment_repo/create_github_repository_job.rb

```diff
@@ -19,6 +19,9 @@ class CreateGitHubRepositoryJob < ApplicationJob
     def perform(assignment, user)
       start = Time.zone.now
 
+      invitation_status = assignment.invitation&.status
+      return if %w[unaccepted creating_repo importing_starter_code completed].include?(invitation_status)
```

Comment:
```ruby
invitation_status = assignment.invitation&.status
return if %w[unaccepted creating_repo importing_starter_code completed].include?(invitation_status)
assignment.invitation&.creating_repo!
```

Should this happen inside an ActiveRecord transaction?

Reply:
No I think we're good, I don't see any reason to have a transaction here